### PR TITLE
PYIC-4124: Added consumer side contract test suite for BAV CRI

### PR DESCRIPTION
**Pull request by RogVanGDS**
### What changed

Added consumer side contract test suite for BAV CRI

### Why did it change

To build up contract test coverage for NINO CRI service, and unblock producer side tests for BAV CRI

### Issue tracking

- [PYIC-4124](https://govukverify.atlassian.net/browse/PYIC-4124)

[PYIC-4124]: https://govukverify.atlassian.net/browse/PYIC-4124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java [hunk 1]
```diff
@@ -0,0 +1,655 @@
...
+              "nbf": 4070908800,
+              "exp": 4070909400,
+              "vc": {
+                "evidence": [
+                  {
+                    "validityScore": 2,
+                    "strengthScore": 3,
+                    "txn": "dummyTxn",
+                    "type": "IdentityCheck"
+                  }
```
Comment (DanCorderIPV) [inline lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java:123]: Where did you find this structure? The stub data has an extra `checkDetails` section and I can't find any documentation in Confluence for the VC structure.

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java [hunk 2]
```diff
@@ -0,0 +1,655 @@
...
+                         "failedCheckDetails": [
+                           {
+                             "identityCheckPolicy": "none",
+                             "checkMethod": "data"
+                           }
+                            ],
+                         "validityScore": 0,
+                         "strengthScore": 3,
+                         "ci": [
+                           "D02"
```
Comment (DanCorderIPV) [inline lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java]: Is `D02` a real CI BAV can return? The stub data has `D15` for BAV

**Review by DanCorderIPV — CHANGES_REQUESTED** [review]
Looks good, but how confident are you that there aren't any other forms of VC that the BAV CRI can return?

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java (hunk 2 above)
Comment (RogVanGDS) [reply lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java]: `D15` is the only CI BAV can return to us, however IPV Core cannot handle `D15` as a CI, it believes that it is invalid.

Currently in the journey map, there is no ability to handle CIs with the BAV_CRI, however the validate() method that we are running in our code believes that `D02` is a valid CI.

Not sure really what the best plan of action for this is, but feels like changing the CI handling is perhaps outside of the scope of this ticket.

@blakeyp any thoughts?

Comment (DanCorderIPV) [reply lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java]: Not sure that's correct, have you tried changing line 626?

Comment (RogVanGDS) [reply lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java]: Oh damn, I forgot we had to mock that out 🤦

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java (hunk 1 above)
Comment (RogVanGDS) [reply lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java:123]: Ah yep I must have missed that section off by mistake

**Comment by RogVanGDS** [comment]
> Looks good, but how confident are you that there aren't any other forms of VC that the BAV CRI can return?

Pretty confident, I've verified the logic with the actual BAV Stub code in their repository.

**Review by DanCorderIPV — APPROVED** [review]

**Comment by blakeyp** [comment]
Yeah a VC with a CI (D15) is the only 'fail' VC we expect to receive from BAV (see RFC https://github.com/govuk-one-login/architecture/blob/8018373649525bbe22e7a5bedafcf4b46c78e50e/rfc/0062-bank-account-verification.md?plain=1#L428)

**Pull request merged**